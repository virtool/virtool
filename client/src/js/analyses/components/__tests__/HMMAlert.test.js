import { AnalysisHMMAlert, mapStateToProps } from "../HMMAlert";

describe("<AnalysisHMMAlert />", () => {
    let props;

    beforeEach(() => {
        props = {
            installed: true
        };
    });

    it("should render", () => {
        const wrapper = shallow(<AnalysisHMMAlert {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [installed = false]", () => {
        props.installed = false;
        const wrapper = shallow(<AnalysisHMMAlert {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps", () => {
    const state = {
        hmms: {
            status: {
                installed: {
                    foo: "bar"
                }
            }
        }
    };
    const props = mapStateToProps(state);

    expect(props).toEqual({
        installed: true
    });
});

describe("mapStateToProps when installed is not defineds", () => {
    const state = {
        hmms: {
            status: {}
        }
    };
    const props = mapStateToProps(state);

    expect(props).toEqual({
        installed: false
    });
});
