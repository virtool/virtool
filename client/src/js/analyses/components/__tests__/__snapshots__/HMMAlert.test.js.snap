// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalysisHMMAlert /> should render 1`] = `""`;

exports[`<AnalysisHMMAlert /> should render when [installed = false] 1`] = `
<Alert__WarningAlert
  level={true}
>
  <Icon
    faStyle="fas"
    fixedWidth={false}
    name="info-circle"
    pullRight={false}
  />
  <span>
    <span>
      HMM data is not installed. 
    </span>
    <Link
      to="/hmm"
    >
      Install HMMs
    </Link>
    <span>
       to run NuV analyses.
    </span>
  </span>
</Alert__WarningAlert>
`;
